okta-1186041-self-host (#6168)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/configure-mcp-authentication/main/index.md

    content: Create an Okta admin app integration and obtain the credentials to connect your Okta MCP server.
---

Create an Okta admin app integration, configure authentication credentials, and grant required OAuth 2.0 scopes to authorize the Model Context Protocol (MCP) server tools.

---

#### Learning outcomes

* Implement a device authorization grant or private key JWT to authorize the Okta MCP server.

> * Get the client ID and security keys and identify the OAuth 2.0 scopes required for your specific use cases.

Reviewer: ```suggestion
* Get the client ID and security keys and identify the OAuth 2.0 scopes that are required for your specific use cases.
```

> * Configure a scope-based tool loading to match your app permissions.

Reviewer: ```suggestion
* Configure scope-based tool loading to match your app permissions.
```

You've created an app integration, configured the authentication flow, and granted the required API scopes for your Okta MCP server.

## Configure scope-based tool loading

> The Okta MCP server uses a scope-based tool loading mechanism to ensure that only the tools your application is authorized to use are exposed to the LLM.

Reviewer: ```suggestion
The Okta MCP server uses a scope-based tool loading mechanism to ensure that only the tools your app is authorized to use are exposed to the LLM.
```

| `okta.templates.manage` | `create_email_customization`, `replace_email_customization`, `delete_email_customization`, `delete_all_email_customizations`, `replace_email_settings`, `send_test_email` |
| `okta.domains.read` | `list_custom_domains`, `get_custom_domain` |
| `okta.domains.manage` | `create_custom_domain`, `replace_custom_domain`, `delete_custom_domain`, `upsert_custom_domain_certificate`, `verify_custom_domain` |
| `okta.emailDomains.read` | `list_email_domains`, `get_email_domain` |
| `okta.emailDomains.manage` | `create_email_domain`, `replace_email_domain`, `delete_email_domain`, `verify_email_domain` |

  > **Note:**  Scopes follow the pattern `okta.<resource>.read` for read-only access and `okta.<resource>.manage` for full read and write access. The `manage` scope implicitly enables all corresponding read operations on that resource.

### Enable a tool group

To activate specific tools, complete the following configuration steps:

1. Grant the scope to your Okta app:

>    1. Open your app integration in the Admin Console.

Reviewer: Use a, b, c for these steps

>    1. Select the Okta API scopes tab and click Grant next to each scope you want to enable.

Reviewer: ```suggestion
   1. Select the **Okta API scopes** tab and click **Grant** for each scope that you want to enable.
```

1. Add the scope to client configurations:

>    1. Update the `OKTA_SCOPES` value in your `mcp.json` or `settings.json` file.

Reviewer: Same comment as above

   1. Include the required scopes as a space-separated string:

   ```
   "OKTA_SCOPES": "okta.users.read okta.groups.read okta.brands.read okta.templates.read"
   ```

1. Restart the client: Restart your MCP client app to apply the updated scope configurations.

>   > **Note:** Start with the minimum set of scopes your use case requires. For example, if you only need to read users and brands, use `okta.users.read okta.brands.read`. Managing scopes enables write operations, so only grant them when required.

Reviewer: ```suggestion
  > **Note:** Start with the minimum set of scopes that your use case requires. For example, if you only need to read users and brands, use `okta.users.read okta.brands.read`. Managing scopes enables write operations, so only grant them when required.
```
